File: frontend/src/app/models/vehicule.ts
export interface Vehicule {
  id?: number;
  matricule: string;
  marque: string;
  modele: string;
  prixParJour: number;
  dateMiseEnService: Date;
  statut: string;
  nombrePortes?: number;
  typeCarburant?: string;
  boiteVitesse?: string;
  cylindree?: number;
  typeMoto?: string;
  casqueInclus?: boolean;
}

File: backend/src/main/java/com/salma/majri/entities/Vehicule.java
package com.salma.majri.entities;

import com.salma.majri.enums.StatutVehicule;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.util.Date;

@Entity
@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
@DiscriminatorColumn(name = "TYPE_VEHICULE", length = 10)
@Data
@NoArgsConstructor
@AllArgsConstructor
public abstract class Vehicule {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String marque;
    private String modele;
    private String matricule;
    private double prixParJour;

    @Temporal(TemporalType.DATE)
    private Date dateMiseEnService;

    @Enumerated(EnumType.STRING)
    private StatutVehicule statut;

    @ManyToOne
    private Agence agence;
}